### PR TITLE
Add mmm-mode and starter config for Markdown parsing

### PR DESCRIPTION
**Pull request by kennethlove**
This isn't ready for merge yet, I know. I'm still fairly new to elisp so I'm sure there are things I'm doing wrong/could be doing better.

We also need more code blocks defined because I'm sure more than just Python people will want this.

### CODE REVIEW

**Comment by kennethlove** [comment]
Handy links:

http://vxlabs.com/2014/04/08/syntax-highlighting-markdown-fenced-code-blocks-in-emacs/

https://github.com/purcell/mmm-mode/blob/master/mmm-sample.el

File: contrib/lang/markdown/packages.el [hunk 1]
```diff
@@ -99,3 +100,22 @@ which require an initialization must be listed explicitly in the list.")
...
+    (mmm-add-classes
+     '((markdown-python
+        :submode python-mode
+        :face mmm-declaration-submode-face
+        :front "^```python[\n\r]+"
+        :back "^```$"))
+     )
+    (setq mmm-global-mode 't)
+    (mmm-add-mode-ext-class 'markdown-mode nil 'markdown-python)
+    (progn
```
Comment (person808) [inline contrib/lang/markdown/packages.el]: Pretty sure this is unneccesary.

File: contrib/lang/markdown/packages.el [hunk 2]
```diff
@@ -99,3 +100,22 @@ which require an initialization must be listed explicitly in the list.")
 (defun markdown/init-markdown-toc ()
   (use-package markdown-toc
     :defer t))
+
+(defun markdown/init-mmm-mode ()
+  (use-package mmm-mode
+    :config
```
Comment (person808) [inline contrib/lang/markdown/packages.el]: I would wrap the whole `:config` block with a `progn` to be consistent with the rest of spacemacs.

**Comment by roryk** [comment]
R would be totally sweet, right now AFAIK there isn't a great solution for writing Rmarkdown with evil-mode on.

**Comment by kennethlove** [comment]
@roryk like the R language? Feel free to submit a PR to my fork w/ R markers.

**Comment by roryk** [comment]
@kennethlove, yup, you got it, R the language. I kind of don't know how to fork your fork to push to without rewriting my fork. Do I just add yours as a remote and pull into a new branch?

**Comment by kennethlove** [comment]
@roryk Maybe? That's what my co-worker thinks. Add mine as a remote, pull my branch, make your changes, push that to your GitHub and then submit that as a PR against mine.


I guess.

**Comment by roryk** [comment]
Thanks, I've never been lead astray following what @kennethlove's coworker says.

**Comment by syl20bnr** [comment]
Never too late to merge :-)
It works like a charm, very handy :+1: 

Thank you !
Cherry-picked into `develop` branch, you can safely delete your branch.

**Pull request closed without merging**